test/unittests/testutil.cc

```diff
@@ -245,7 +247,55 @@ MockCatalog* MockCatalog::AttachFreely(const std::string  &root_path,
   return new_catalog;
 }
 
-void MockCatalog::RegisterChild(MockCatalog *child) {
+void MockCatalog::RemoveChild(MockCatalog *child) {
+  std::vector<NestedCatalog>::iterator iter = active_children_.begin();
+  bool found = false;
+  while (!found && iter != active_children_.end()) {
+    if (iter->hash == child->hash()) {
+      active_children_.erase(iter);
+      found = true;
+    }
+    iter++;
+  }
+}
+
+MockCatalog* MockCatalog::FindSubtree(const PathString &path) const {
+  for (unsigned i = 0; i < active_children_.size(); ++i) {
+    if (active_children_[i].path == path)
+      return active_children_[i].child;
+  }
+  return NULL;
+}
+
+bool MockCatalog::LookupPath(const PathString &path,
+                catalog::DirectoryEntry *dirent) const {
+  shash::Md5 md5_path(path.GetChars(), path.GetLength());
+  for (unsigned i = 0; i < files_.size(); ++i) {
+    if (files_[i].path_hash == md5_path) {
+      *dirent = files_[i].ToDirectoryEntry();
+      return true;
+    }
+  }
+  return false;
+}
+
+bool MockCatalog::ListingPath(const PathString &path,
+                 catalog::DirectoryEntryList *listing) const {
+  listing->clear();
+  bool found = false;
+  for (unsigned i = 0; i < files_.size(); ++i) {
+    shash::Md5 path_hash(path.GetChars(), path.GetLength());
+    if (files_[i].parent_hash == path_hash)
+      listing->push_back(files_[i].ToDirectoryEntry());
+    else if (files_[i].path_hash == path_hash)
+      found = true;
+  }
+  if (!found)  // the parent does not exists, nor the children then
```

Comment:
this is not necessary, is it?  I think the real catalog would return children even if the parent is missing.